synapse/api/auth.py

```diff
@@ -629,7 +629,13 @@ def get_user_by_access_token(self, token, rights="access"):
         except AuthError:
             # TODO(daniel): Remove this fallback when all existing access tokens
             # have been re-issued as macaroons.
+            if self.hs.config.expire_access_token:
+                raise AuthError(
+                    401,
+                    "access token has expired"
+                )
```

Comment:
I've taken a look at the macaroon stuff and it doesn't appear to be easy to tell which caveats have failed. I'd be sorely tempted to just reraise the exception if `config.expire_access_token` is enabled (which would then effectively disable support for creating non-macaroon access tokens). i.e.:

```python
except AuthError:
    if self.hs.config.expire_access_token:
        raise
    ret = yield self._look_up_user_by_access_token(token)